### ISSUE TITLE
Error while registering users and organization also unable to find service in marketplace

### ISSUE THREAD

**Issue by jarunkumarb** [servicecatalog/oscm#477]
**Version Info**
v18

Hi sir,
   Thank you very much for your great support.
   We followed all the  steps what you have given.
   All are working fine.
   Still we are facing few issues ie listed below.
   Please help me as soon as possible.

 This is my var.env file configurations please let me know what are the changes to be done or is anything missing.

#####################
# Database settings #
#####################

DB_PORT_CORE=5432
DB_PORT_JMS=5432
DB_PORT_APP=5432

DB_PWD_CORE=bssuser

DB_PWD_APP=bssappuser

DB_SUPERPWD=postgres

#################
# Mail settings #
#################
SMTP_HOST=smtp.gmail.com

SMTP_PORT=587

SMTP_FROM=myemail@gmail.com

SMTP_USER=none

SMTP_PWD=mypassword

SMTP_AUTH=true

SMTP_TLS=true

APP_ADMIN_MAIL_ADDRESS=myemail@gmail.com

APP_TRUSTSTORE_BSS_ALIAS=1

#######################
# Controller settings #
#######################

CONTROLLER_ORG_ID=PLATFORM_OPERATOR

CONTROLLER_USER_KEY=1000

CONTROLLER_USER_NAME=administrator

CONTROLLER_USER_PASS=admin123

##############################
# VMware controller settings #
##############################
# Database user settings
# Database user name
# Please set to vmwareuser
#DB_USER_VMWARE=${DB_USER_VMWARE}
# Database user password
#DB_PWD_VMWARE=${DB_PWD_VMWARE}
# VCenter settings
#VCENTER_NAME=${VCENTER_NAME}
# Default name of the VMware default Data Center
#DATACENTER_NAME=${DATACENTER_NAME}
# Default name of the VMware Cluster
#CLUSTER_NAME=${CLUSTER_NAME}
# Default name of the VMware network
#LOAD_BALANCER_NAME=${LOAD_BALANCER_NAME}

#################
# Misc settings #

KEY_SECRET=bss

REPORT_ENGINEURL=https://172.16.1.178:8681/birt/frameset?__report=\${reportname}.rptdesign&SessionId=\${sessionid}&__locale=

\${locale}&WSDLURL=\${wsdlurl}&SOAPEndPoint=\${soapendpoint}&wsname=Report&wsport=ReportPort

TOMEE_DEBUG=true

LOG_LEVEL=INFO

##################
# Tomee settings #
##################

CONTAINER_CALLBACK_THREADS=50

CONTAINER_MAX_SIZE=50
PROXY_ENABLED=${PROXY_ENABLED}
PROXY_HTTP_HOST=${PROXY_HTTP_HOST}
PROXY_HTTP_PORT=${PROXY_HTTP_PORT}
PROXY_HTTPS_HOST=${PROXY_HTTPS_HOST}
PROXY_HTTPS_PORT=${PROXY_HTTPS_PORT}

PROXY_NOPROXY=${PROXY_NOPROXY}

AUTH_MODE=INTERNAL
ADMIN_USER_ID=administrator

######################
# OpenStack settings #
######################
# Keystone URL, e.g. http://10.140.1.1:5000/v3/
# OS_KEYSTONE_URL=
# Login name used for service provisioning
# OS_USER=
# Password for this user
# OS_PASSWORD=
# Name of Domain, e.g. Default
# OS_DOMAIN=
# Default tenant ID used for service provisioning
# OS_TENANT_ID=                       

we are facing issues in following steps

Issue-1:
 Registering new users
	Go to Account->Register New Users
	Here We are Facing the issue Operation Failed.
	**Here newly register user also Not Getting Email.**

Issue-2:
 We try to create organization from the following steps
	Go to service catalog Manager->Under operation->create organization
	**Here We are Facing the issue operation Failed** 

Issue-3:
 we try to subscribe a service from the following steps
	Get it now->configuration->payment->confirmation
	here we are facing the issue The technical service for the subscription returned: **Synchronous provisioning operations are not supported by the asynchronous provisioning proxy.**

Issue-4:
 where to find the service provisioning service xml file to enable the service(User management,authentication,service management,Event management) because am unable to find these services in the marketplace.

**Screenshots**
![regerror](https://user-images.githubusercontent.com/13447493/66207739-2bdd7d80-e6d1-11e9-911a-86e994a2bf68.png)

![payment issue](https://user-images.githubusercontent.com/13447493/66207753-339d2200-e6d1-11e9-958e-ec3b512a89fc.png)

**Comment by GoebelL** [comment]
Hello again, 

Please don't post same issues several times. Points 1 and 2 are duplicate with #429, #426, #473 and #470. 
As mentioned, the details of "Operation Failed" can be found in system.log file. 
```
docker exec -it oscm-core bash
cat /logs/system.log
```
If mail transport is not working, you'll most likely find the cause in this file. Locate the error entry that matches the timestamp of the occurence. 
Despite of this, if you are using google mail, you need to give some additional setting in the google account in order to make it trust the OSCM application.
Login to Gmail, goto [_Account Security Settings_](
https://www.google.com/settings/security/lesssecureapps) and enable _Access for less secure apps_ 
 to make google smtp usable with other clients then the out-of-the-box supported.

Btw, in case you have longer attachments like here, please put them in a single zip file and drop this in the description.

**Comment by GoebelL** [comment]
Issue-3
The error message already says it. You have defined a service with synchronous provisioning. APP is asynchonous (APP = "Asynchronous Provisioning Proxy"). Therefore it doesn't support this mode. I don't know whether this technical service template of your "example 5" service is self-created or imported. But definetly you'll need to set `provisioningType="ASYNCHRONOUS"` in all services to make them work with APP. Refer to chapter 3.1 and following in the [Technology Provider Guide](https://github.com/servicecatalog/documentation/blob/master/Development/oscm-doc-user/resources/manuals/integration/en/TechProv.pdf).

issue-4
Not clear which services you expect to find in the marketplace. It seems to me that this got a bit mixed: API services, like the [ServiceProvisioningService](https://github.com/servicecatalog/oscm-interfaces/blob/master/oscm-extsvc/src/main/java/org/oscm/intf/ServiceProvisioningService.java) interface and marketabe services in the application, which represent the customer offerings that the service manager defines in the OSCM portal and publishes in the marketplace. Two very different things. 
Basically the technology manager provides the technical services, the service manager creates the marketable services with or without prices, defines editions, upgrade and downgrade options with our without pricemodels and activates them. Once they are active they get visible in the marketplace, where they are published. 
Events are provided by the technical service and described in the technical service template. The supplier can choose to define prices for them. More details [here](https://github.com/servicecatalog/documentation/blob/master/Development/oscm-doc-user/resources/manuals/integration/en/Supplier.pdf).

**Comment by GoebelL** [comment]
@jarunkumarb 
I'm looking in the var.env settings you posted above. There might be some more to be corrected.

1. With gmail you need SMTP_AUTH=true and provide credentials of your google account. Of course you don't post this here, but note that SMTP_USER=none is for using no authentication and probably not the right user id in your case. 

2. Are your proxy settings resolving correctly? As described in the comment at beginning of the file, all placeholders need to be replaced with values (unless you have defined them elsewhere).
Here following settings starting with "${PROXY_" are not replaced: PROXY_ENABLED=${PROXY_ENABLED}
PROXY_HTTP_HOST=${PROXY_HTTP_HOST}
PROXY_HTTP_PORT=${PROXY_HTTP_PORT}
PROXY_HTTPS_HOST=${PROXY_HTTPS_HOST}
PROXY_HTTPS_PORT=${PROXY_HTTPS_PORT}
PROXY_NOPROXY=${PROXY_NOPROXY}

**Comment by GoebelL** [comment]
This article describes using Java applications with gmail 
https://www.mkyong.com/java/javamail-api-sending-email-via-gmail-smtp-example/
Required steps should be same for OSCM with gmail.

**Comment by jarunkumarb** [comment]
Thank you very many much sir. Sorry to disturbing you.
